test/tests/functional/pbs_hook_set_jobenv.py

```diff
@@ -0,0 +1,887 @@
+                         self.env_nohook['TEST_DQUOTE4'].rstrip('\n'))
+        self.logger.info("TEST_DQUOTE4 - " + os.environ['TEST_DQUOTE4'] +
+                         " == " + self.env_nohook['TEST_DQUOTE4'].rstrip('\n'))
+        self.assertEqual(os.environ['TEST_DQUOTE5'],
+                         self.env_nohook['TEST_DQUOTE5'].rstrip('\n'))
+        self.logger.info("TEST_DQUOTE5 - " + os.environ['TEST_DQUOTE5'] +
+                         " == " + self.env_nohook['TEST_DQUOTE5'].rstrip('\n'))
+        self.assertEqual(os.environ['TEST_DQUOTE6'],
+                         self.env_nohook['TEST_DQUOTE6'].rstrip('\n'))
+        self.logger.info("TEST_DQUOTE6 - " + os.environ['TEST_DQUOTE6'] +
+                         " == " + self.env_nohook['TEST_DQUOTE6'].rstrip('\n'))
+        self.assertEqual(os.environ['TEST_SQUOTE'],
+                         self.env_nohook['TEST_SQUOTE'].rstrip('\n'))
+        self.logger.info("TEST_SQUOTE - " + os.environ['TEST_SQUOTE'] +
+                         " == " + self.env_nohook['TEST_SQUOTE'].rstrip('\n'))
+        self.assertEqual(os.environ['TEST_SQUOTE2'],
+                         self.env_nohook['TEST_SQUOTE2'].rstrip('\n'))
+        self.logger.info("TEST_SQUOTE2 - " + os.environ['TEST_SQUOTE2'] +
+                         " == " + self.env_nohook['TEST_SQUOTE2'].rstrip('\n'))
+        self.assertEqual(os.environ['TEST_SQUOTE3'],
+                         self.env_nohook['TEST_SQUOTE3'].rstrip('\n'))
+        self.logger.info("TEST_SQUOTE3 - " + os.environ['TEST_SQUOTE3'] +
+                         " == " + self.env_nohook['TEST_SQUOTE3'].rstrip('\n'))
+        self.assertEqual(os.environ['TEST_SQUOTE4'],
+                         self.env_nohook['TEST_SQUOTE4'].rstrip('\n'))
+        self.logger.info("TEST_SQUOTE4 - " + os.environ['TEST_SQUOTE4'] +
+                         " == " + self.env_nohook['TEST_SQUOTE4'].rstrip('\n'))
+        self.assertEqual(os.environ['TEST_SQUOTE5'],
+                         self.env_nohook['TEST_SQUOTE5'].rstrip('\n'))
+        self.logger.info("TEST_SQUOTE5 - " + os.environ['TEST_SQUOTE5'] +
+                         " == " + self.env_nohook['TEST_SQUOTE5'].rstrip('\n'))
+        self.assertEqual(os.environ['TEST_SQUOTE6'],
+                         self.env_nohook['TEST_SQUOTE6'].rstrip('\n'))
+        self.logger.info("TEST_SQUOTE6 - " + os.environ['TEST_SQUOTE6'] +
+                         " == " + self.env_nohook['TEST_SQUOTE6'].rstrip('\n'))
+        self.assertEqual(os.environ['BROL'],
+                         self.env_nohook['BROL'].rstrip('\n'))
+        self.logger.info("BROL - " + os.environ['BROL'] + " == " +
+                         self.env_nohook['BROL'].rstrip('\n'))
+        self.assertEqual(os.environ['BROL1'],
+                         self.env_nohook['BROL1'].rstrip('\n'))
+        self.logger.info("BROL - " + os.environ['BROL1'] + " == " +
+                         self.env_nohook['BROL1'].rstrip('\n'))
```

Comment:
Isn't this the same as common_validate()?

Reply:
Similar but not quite the same.